CONTRIBUTING.md

```diff
@@ -74,9 +74,57 @@ We enjoy working with contributors to get their code accepted. There are many ap
 * Feel free to make as many commits as you want, while working on a branch.
 * When submitting a PR for review, please perform an interactive rebase to present a logical history that's easy for the reviewers to follow.
 * Please use your commit messages to include helpful information on your changes, e.g. changes to APIs, UX changes, bugs fixed, and an explanation of *why* you made the changes that you did.
-* Resolve merge conflicts by rebasing the target branch over your feature branch, and force-pushing.
+* Resolve merge conflicts by rebasing the target branch over your feature branch, and force-pushing (see below for instructions).
 * When merging, we'll squash your commits into a single commit.
 
+#### Rebasing and fixing merge conflicts
+
+Rebasing can be tricky, and fixing merge conflicts can be even trickier because it involves force pushing. This is all compounded by the fact that attempting to push a rebased branch remotely will be rejected by git, and you'll be prompted to do a `pull`, which is not at all what you should do (this will really mess up your branch's history).
+
+Here's how you should rebase master onto your branch, and how to fix merge conflicts when they arise.
+
+First, make sure master is up-to-date.
+
+```
+git checkout master
+git fetch upstream
```

Comment:
This is the first time this doc mentions `upstream`, we should probably explain what it is (if we already need this to explain people git)

Also, I usually `git pull --rebase upstream master`, but two commands is probably okey ;)